common/elf/analyzer.go

```diff
@@ -43,6 +43,17 @@ type ElfAnalyzer struct {
 
 var ErrSymbolNotFound = errors.New("symbol not found")
 
+// HasElfMagic checks if the file starts with an ELF magic number.
+func HasElfMagic(bytesArray []byte) bool {
```

Comment:
I am not sure if I agree with the change of name from `IsElf`.
The fact that the logic is checking the ELF magic is just that - implementation detail.
This is a very normal ELF check, so calling it by it's purpose is clearer in my opinion than calling it by the implementation of it.

Reply:
I'm not strong opinionated with this, however:

- `HasElfMagic()` implicitly communicates that it checks only for ELF magic bytes.
- `IsElf()` implicitly communicates that it might check the whole file for ELF structures and so.

Reply:
This is what it does now, and this function name reflects it. If we will add more checks we can add another function that can call this one, and export it as well